plugins/modules/ec2_metadata_facts.py

```diff
@@ -484,11 +487,47 @@ def __init__(
         self._token = None
         self._prefix = "ansible_ec2_%s"
 
+    def decode_user_data(self, data):
+        is_compressed = False
+
+        # Get the current locale's encoding
+        current_encoding = locale.getpreferredencoding()
```

Comment:
I'm not sure about this. We don't know what the user data was encoded as, and utf-8 is probably a better guess than locale encoding.